packages/cinterop/src/commonMain/kotlin/io/realm/interop/Errors.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.interop
+
+/**
+ * Generic representation of a Realm-Core exception.
+ */
+sealed class RealmCoreException(message: String?) : Exception(message)
```

Comment:
Don't know if it is strictly needed, but I would be somehow more comfortable if we actually included the error id of the core error somewhere in the `RealmCoreException`. But if the enum mappings are actually validated in the test, then it is maybe ok to leave out. Don't know if other share the same gut feeling though.

Reply:
I was thinking the same thing...but as long as we have automation in place for making sure that we don't skip mapping some codes, then it is probably safe, and the enums provide more immediate context.

Reply:
We should be safe as we have exhaustive mapping tests. I am going to check if we can append the error code into the message.

Reply:
I have reverted to the previous messaging: [$errorcode]: $message.